Add bug form

juancarlospaco opened the pull request:
- Continuation of https://github.com/nim-lang/RFCs/pull/459#issue-1227097047
- No code in this PR just GitHub repo metadata stuff.
- Form for bug reports and feature requests. 
- Users can fill in information about OS, Hardware, etc that can be useful for devs (users can choose "Unknown" too).
- See how it works at https://github.com/nim-lang/RFCs/issues/new?assignees=&labels=unconfirmed&template=bug_report.yml


I changed the existing template in the repo to a GitHub issue form, just 2 YAML.

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

Just improving the UX and making things easier for the users. 🙂

juancarlospaco committed:
Add Bug Form

juancarlospaco committed:
Add Bug Form

juancarlospaco committed:
Add Bug Form

juancarlospaco committed:
Add Bug Form

ringabout commented:
Thanks for your hard work! I think there are too many required selections. Collecting info regarding "Where are you from" doesn't seem necessary (anyway, using continent instead of country and Region could be better).

juancarlospaco committed:
https://github.com/nim-lang/Nim/pull/19913#issuecomment-1160663243

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,430 @@
+- type: markdown
+  attributes:
+    value: |
+      - **Please provide a minimal code example that reproduces the Bug!.** :bug:
```
```suggestion
      - **Please provide a minimal code example that reproduces the Bug!** :bug:
```
In general, there is no `.` after a `!` or a `?`.

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,430 @@
+name:        "Bug Report"
+description: "Create a new Bug report. Have you found an unexpected behavior?, use this form."
```
```suggestion
description: "Create a new Bug report. Have you found an unexpected behavior? Use this form."
```
`?` and `!` mark the end of a sentence, so it's incorrect to continue with a `,` afterwards.

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,430 @@
+  attributes:
+    value: |
+      - **Please provide a minimal code example that reproduces the Bug!.** :bug:
+        Reports with full repro code and descriptive detailed information will likely receive fixes faster.
```
```suggestion
        Reports with a reproducible example and descriptive detailed information will likely receive fixes faster.
```

juancarlospaco commented:
> Collecting info regarding "Where are you from" doesn't seem necessary.

I prefer to ask the user (if they want to reply) than adding user-tracking to Nim:
- https://github.com/dom96/choosenim/issues/144#issue-513003732
- https://github.com/nim-lang/RFCs/issues/374#issue-873734945

juancarlospaco committed:
https://github.com/nim-lang/Nim/pull/19913#issuecomment-1160663243

konsumlamm commented:
> Thanks for your hard work! I think there are too many required selections. Collecting info regarding "Where are you from" doesn't seem necessary (anyway, using continent instead of country and Region could be better).

I agree. I don't see how disk storage, memory capacity, CPU cores, internet connection, web browser are relevant either. For most bug reports, it just creates useless clutter in the issue report and in the few cases where that information is relevant, I think we can expect the creator to include that information.

Yardanico commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,223 @@
+name:        "Bug Report"
+description: "Create a new Bug report. Have you found an unexpected behavior?, use this form."
```
```suggestion
description: "Create a new bug report. Have you found an unexpected behavior?, use this form."
```

I don't think we need random capital words

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,223 @@
+  id: incompatibility
+  attributes:
+    label: Backwards Compatibility
+    description: If your Bug Report introduces backward-incompatible changes, describe them and propose how to deal with them.
```
How can a report introduce changes? Do you mean the proposed fix?

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,223 @@
+  attributes:
+    value: |
+      - Thanks for your contributions!, your Bug report will receive feedback from the community soon...
+      - **Remember to :star: Star the Nim project on GitHub!.**
```
This feels a bit needy, I don't think it's appropriate for the bug report form (it's like everyone says "please like and subscribe", it's not very sympathetic). If people feel like Nim deserves a star, they will star it, otherwise they don't.

juancarlospaco replied:
It was already on https://github.com/nim-lang/RFCs/pull/459#issue-1227097047  🤷

konsumlamm commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,223 @@
+    value: |
+      - Thanks for your contributions!, your Bug report will receive feedback from the community soon...
+      - **Remember to :star: Star the Nim project on GitHub!.**
+      - Please check whether the problem still exists in git devel branch, see [rebuilding the compiler](https://nim-lang.github.io/Nim/intern.html#rebuilding-the-compiler).
```
```suggestion
      - Please check whether the problem still exists in the devel branch, see [rebuilding the compiler](https://nim-lang.github.io/Nim/intern.html#rebuilding-the-compiler).
```
Maybe suggest `choosenim devel --latest`, that's a lot easier imo.

konsumlamm commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,73 @@
+- type: markdown
+  attributes:
+    value: |
+      - Please provide a minimal code example that illustrates the basic idea behind your Feature request.
```
```suggestion
      - Please provide a minimal code example that illustrates the basic idea behind your feature request.
```

konsumlamm commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,73 @@
+    value: |
+      - Please provide a minimal code example that illustrates the basic idea behind your Feature request.
+        Reports with full repro code and descriptive detailed information will likely receive feedback faster.
+      - There is a separate repository for the detailed RFCs and proposals: https://github.com/nim-lang/RFCs
```
```suggestion
      - There is a separate repository for the detailed RFCs and proposals: https://github.com/nim-lang/RFCs.
```

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: Yardanico <tiberiumk12@gmail.com>

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/bug_report.yml

Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/feature_request.yml

Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>

juancarlospaco committed:
Update .github/ISSUE_TEMPLATE/feature_request.yml

Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>

juancarlospaco committed:
Do not Star Nim on Github

juancarlospaco committed:
No backwards compat problems

juancarlospaco committed:
Typo

juancarlospaco commented:
> I think we can expect the creator to include that information.

I think the problem usually is that bug reports have too scarce information,
bug reports with too much information is not a very frequent problem.

As a dev myself I think it is very useful,
example if the user says uses Linux, you can just reply with Bash commands to fix it,
if the user says 32bit Windows maybe the hardware is too old for the software, etc,
gives more context of the problem, instead of the typical "shit wont work lol" kind of reports.

You can just not fill it in anyway... 🤷

Araq commented:
> You can just not fill it in anyway... 🤷

You can also just not fill in a bug report if the task appears to be overwhelming...

juancarlospaco commented:
> You can also just not fill in a bug report

I agree, sending a PR instead is also an alternative, as the current template suggests too.

@Araq
Let me know what do you want to remove or add to the form.

juancarlospaco committed:
Merge branch 'devel' into bugform

Araq commented:
I don't understand it, your screenshot is about RFCs, not about bug reports.

juancarlospaco commented:
> your screenshot is about RFCs, not about bug reports.

True, I removed the screenshots.

It is similar to the one on the RFC repo, but for this repo,
I added fields to the form for info that may be useful for devs to get more context of the bug,
but it is easy to remove the fields, no problem,
it is flexible to whatever you think a bug report form should look like,
what do you want to know when you read a bug report ?

ringabout commented:
Here is the preview

https://github.com/nim-lang/Nim/blob/184b5abb92d5d40b886694d92110cebc17d8276d/.github/ISSUE_TEMPLATE/bug_report.yml

https://github.com/nim-lang/Nim/blob/184b5abb92d5d40b886694d92110cebc17d8276d/.github/ISSUE_TEMPLATE/feature_request.yml

juancarlospaco commented:
> Here is the preview

Thanks @xflywind thats a lot more direct lol 🙂:+1:

juancarlospaco committed:
Merge branch 'devel' into bugform

juancarlospaco committed:
Merge branch 'devel' into bugform

juancarlospaco committed:
Merge branch 'devel' into bugform

juancarlospaco committed:
Merge branch 'devel' into bugform

juancarlospaco committed:
Merge branch 'devel' into bugform

juancarlospaco commented:
@Araq 
Any blockers to merge this?, it is not even code, so should be easy to merge,
if some field needs to be removed let me know, any user can improve it later anyways.
